include/xtensor-signal/lfilter.hpp

```diff
@@ -0,0 +1,92 @@
+#ifndef XTENSOR_SIGNAL_LFILTER_HPP
+#define XTENSOR_SIGNAL_LFILTER_HPP
+
+#include <type_traits>
+#include <xtensor-blas/xlinalg.hpp>
+#include <xtensor/xarray.hpp>
+#include <xtensor/xaxis_slice_iterator.hpp>
+#include <xtensor/xbuilder.hpp>
+#include <xtensor/xexception.hpp>
+#include <xtensor/xindex_view.hpp>
+#include <xtensor/xnoalias.hpp>
+#include <xtensor/xpad.hpp>
+#include <xtensor/xtensor.hpp>
+
+namespace xt {
+namespace signal {
+namespace detail {
+template <typename E1, typename E2, typename E3, typename E4>
+inline auto lfilter(E1 &&b, E2 &&a, E3 &&x, E4 zi) {
+  using value_type = typename std::decay_t<E3>::value_type;
+  if (zi.shape(0) != x.shape(0)) {
+    XTENSOR_THROW(
+        std::runtime_error,
+        "Accumulator initialization must be the same length as the input");
+  }
+  if (x.dimension() != 1) {
+    XTENSOR_THROW(std::runtime_error,
+                  "Implementation only works on 1D arguments");
+  }
+  if (a.dimension() != 1) {
+    XTENSOR_THROW(std::runtime_error,
+                  "Implementation only works on 1D arguments");
+  }
+  if (b.dimension() != 1) {
+    XTENSOR_THROW(std::runtime_error,
+                  "Implementation only works on 1D arguments");
+  }
+  xt::xtensor<value_type, 1> out = xt::zeros_like(x);
+  for (int i = 0; i < x.shape(0); i++) {
+    auto padded_x = xt::pad(x, b.shape(0) - 1);
+    auto b_accum =
+        xt::sum(b *
+                xt::flip(xt::view(padded_x, xt::range(i, i + b.shape(0))))) +
+        zi(i);
+    auto padded_out = xt::pad(out, a.shape(0) - 1);
```

Comment:
Same here.

Reply:
I will move this out too but needs to be more clever as output is updating with each iteration.